src/PrestaShopBundle/Controller/Admin/Sell/Order/InvoicesController.php

```diff
@@ -0,0 +1,223 @@
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash(
+                'error',
+                $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error')
+            );
+
+            return $this->redirectToRoute('admin_order_invoices');
+        }
+
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_date.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            } else {
+                $invoiceDataProvider = $this->get('prestashop.adapter.data_provider.order_invoice');
+
+                // Get invoices by submitted date interval
+                $invoiceCollection = $invoiceDataProvider->getByDateInterval(
+                    $data['generate_by_date']['date_from'],
+                    $data['generate_by_date']['date_to']
+                );
+
+                if (empty($invoiceCollection)) {
+                    $this->addFlash(
+                        'error',
+                        $this->trans('No invoice was found.', 'Admin.Orderscustomers.Notification')
+                    );
+                } else {
+                    // Generate PDF out of found invoices
+                    $invoiceGenerator = $this->get('prestashop.adapter.pdf_generator.invoice');
+                    $invoiceGenerator->generateInvoicesPDF($invoiceCollection);
+                }
+            }
+        }
+
+        return $this->redirectToRoute('admin_order_invoices');
+    }
+
+    /**
+     * Action that generates invoices PDF by order status
+     *
+     * @param Request $request
+
+     * @return RedirectResponse
+     */
+    public function generatePdfByStatusAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if (!in_array(
+            $this->authorizationLevel($legacyController),
+            [
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash(
+                'error',
+                $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error')
+            );
+
+            return $this->redirectToRoute('admin_order_invoices');
+        }
+
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_status.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            } else {
```

Comment:
same here, everything in the `else` statement should be part of the form handler as you are trying to handle form data in the controller :)